core/src/test/java/org/apache/iceberg/view/ViewCatalogTests.java

```diff
@@ -0,0 +1,876 @@
+    assertThat(view.name()).isEqualTo(catalog().name() + "." + identifier);
+    assertThat(view.properties()).isEqualTo(ImmutableMap.of("prop1", "val1", "prop2", "val2"));
+    assertThat(view.history())
+        .hasSize(1)
+        .first()
+        .extracting(ViewHistoryEntry::versionId)
+        .isEqualTo(1);
+    assertThat(view.schema().schemaId()).isEqualTo(SCHEMA.schemaId());
+    assertThat(view.schema().asStruct()).isEqualTo(SCHEMA.asStruct());
+    assertThat(view.schemas()).hasSize(1).containsKey(SCHEMA.schemaId());
+    assertThat(view.versions()).hasSize(1).containsExactly(view.currentVersion());
+
+    ViewVersion viewVersion = view.currentVersion();
+    assertThat(viewVersion).isNotNull();
+    assertThat(viewVersion.versionId()).isEqualTo(1);
+    assertThat(viewVersion.schemaId()).isEqualTo(SCHEMA.schemaId());
+    assertThat(viewVersion.summary()).hasSize(1).containsEntry("operation", "create");
+    assertThat(viewVersion.operation()).isEqualTo("create");
+    assertThat(viewVersion.defaultNamespace()).isEqualTo(identifier.namespace());
+    assertThat(viewVersion.defaultCatalog()).isNull();
+    assertThat(viewVersion.representations())
+        .hasSize(2)
+        .containsExactly(
+            ImmutableSQLViewRepresentation.builder()
+                .sql("select * from ns.tbl")
+                .dialect("spark")
+                .build(),
+            ImmutableSQLViewRepresentation.builder()
+                .sql("select * from ns.tbl using X")
+                .dialect("trino")
+                .build());
+
+    assertThat(catalog().dropView(identifier)).isTrue();
+    assertThat(catalog().viewExists(identifier)).as("View should not exist").isFalse();
+  }
+
+  @Test
+  public void createViewThatAlreadyExists() {
+    TableIdentifier tableIdentifier = TableIdentifier.of("ns", "table");
+    TableIdentifier viewIdentifier = TableIdentifier.of("ns", "view");
+
+    if (requiresNamespaceCreate()) {
+      catalog().createNamespace(viewIdentifier.namespace());
+    }
+
+    assertThat(catalog().viewExists(viewIdentifier)).isFalse();
+
+    View view =
+        catalog()
+            .buildView(viewIdentifier)
+            .withSchema(SCHEMA)
+            .withDefaultNamespace(viewIdentifier.namespace())
+            .withQuery("spark", "select * from ns.tbl")
+            .create();
+
+    assertThat(view).isNotNull();
+    assertThat(catalog().viewExists(tableIdentifier)).isFalse();
+    assertThat(catalog().viewExists(viewIdentifier)).isTrue();
+
+    Assertions.assertThatThrownBy(
+            () ->
+                catalog()
+                    .buildView(viewIdentifier)
+                    .withSchema(OTHER_SCHEMA)
+                    .withQuery("spark", "select * from ns.tbl")
+                    .create())
+        .isInstanceOf(AlreadyExistsException.class)
+        .hasMessageStartingWith("View already exists: ns.view");
+
+    catalog().buildTable(tableIdentifier, SCHEMA).create();
```

Comment:
This test case is doing too much. This should test just the behavior of the view catalog, not the behavior when a catalog supports both views and tables. Using a separate test case for additional tests when a catalog supports tables is a good idea because it also allows us to turn off the tests when working with only a view catalog.

In addition, I think that a result of testing this way (focusing on `viewExists`) results in missing test cases that should be here. For example, `tableExists` should have a similar test that it does not recognize views.

Reply:
I've split this out into separate tests and also added a test that verifies you can't create a table if a view with the same name already exists